File: src/screens/onboarding/ReminderOnboarding.js
import { View, Text, StyleSheet } from 'react-native'
import React, { useState } from 'react'
import colors from '../../utilities/colors';
import { SafeAreaView } from 'react-native-safe-area-context';
import SwichableButton from '../../components/settingScreen/SwichableButton';
import TimePicker from '../../components/settingScreen/TimePicker';
import ArrowButton from '../../components/onboarding/ArrowButton';
import BackButton from '../../components/onboarding/BackButton';

export default function ReminderOnboarding({ onNext, onBack, setUserData }) {
    const [ReminderIsEnabled, setReminderIsEnabled] = useState(true)
    const onPress = () => {
        setUserData(prev => ({ ...prev, reminder: ReminderIsEnabled }));
        onNext()
    }
    const timeHandler = (time) => {
        setUserData(prev => ({
            ...prev,
            reminderHours: time.hours,
            reminderMinutes: time.minutes
        }));

    }
    return (
        <SafeAreaView style={styles.container} >
            <View>
                <BackButton onPress={onBack} />
            </View>
            <View style={styles.content}>
                <Text style={styles.title}>Set Therapy Reminder</Text>
                <Text style={styles.subtitle}>
                    In order not to disrupt the daily therapy process, we will send you a reminder half an hour before the time you set.
                </Text>
                <SwichableButton isEnabled={ReminderIsEnabled} setIsEnabled={setReminderIsEnabled} title={'Enable Therapy Reminder'} />

                {ReminderIsEnabled && <>
                    <Text style={styles.subtitle}>
                        Schedule a meeting with your therapist
                    </Text>
                    <TimePicker timeHandler={timeHandler} /></>}
            </View>
            <View style={styles.console}>
                <ArrowButton onPress={onPress} />
            </View>
        </SafeAreaView>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: colors.black,
        paddingTop: 20,
        paddingHorizontal: 20,
    },
    content: {
        flex: 1,
    },
    title: {
        color: colors.white,
        fontWeight: 'bold',
        marginBottom: 10,
        fontSize: 24,
    },
    subtitle: {
        color: 'gray',
        marginBottom: 20,
        fontSize: 16,
        lineHeight: 24,
    },
    console: {
        alignItems: 'flex-end',
        marginBottom: 30,
    },
});

File: lib/screens/onboarding/reminder_onboarding.dart
import 'package:flutter/material.dart';
import 'package:hexcolor/hexcolor.dart';
import 'package:provider/provider.dart';
import 'package:zihnai/generated/l10n.dart';
import 'package:zihnai/ultils/providers/user_provider.dart';
import 'package:zihnai/ultils/constant/color.dart';
import 'package:zihnai/widgets/wheel_time_picker.dart';

class ReminderOnboarding extends StatefulWidget {
  const ReminderOnboarding({
    super.key,
    required this.onNext,
    required this.onBack,
  });
  final void Function() onNext;
  final void Function() onBack;

  @override
  ReminderOnboardingState createState() => ReminderOnboardingState();
}

class ReminderOnboardingState extends State<ReminderOnboarding> {
  @override
  Widget build(BuildContext context) {
    void onPress() {
      widget.onNext();
    }

    void setHours(int hours) {
      context.read<UserProvider>().setReminderHours(hours);
    }

    void setMinutes(int minutes) {
      context.read<UserProvider>().setReminderMinutes(
            minutes,
          );
    }

    return SafeArea(
      child: Scaffold(
        backgroundColor: HexColor(dark),
        body: Padding(
          padding: const EdgeInsets.symmetric(horizontal: 20.0, vertical: 20.0),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              // Back Button
              BackButton(
                onPressed: widget.onBack,
                color: Colors.white,
              ),

              // Content Section
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(
                      S.of(context).onboardingReminderTitle,
                      style: TextStyle(
                        color: Colors.white,
                        fontWeight: FontWeight.bold,
                        fontSize: 24,
                      ),
                    ),
                    Padding(
                      padding: EdgeInsets.symmetric(vertical: 20),
                      child: Text(
                        S.of(context).onboardingReminderSubtitle,
                        style: TextStyle(
                          color: Colors.grey,
                          fontSize: 16,
                        ),
                      ),
                    ),

                    // Switchable Button (SwitchListTile)
                    Container(
                      height: 100,
                      alignment: Alignment.center,
                      decoration: BoxDecoration(
                          color: HexColor(secondary),
                          borderRadius: BorderRadius.circular(12)),
                      child: SwitchListTile(
                        title: Text(
                          S.of(context).reminderButtonTitle,
                          style: TextStyle(color: Colors.white, fontSize: 15),
                        ),
                        value: context.watch<UserProvider>().reminder,
                        onChanged: (bool value) {
                          context.read<UserProvider>().setReminder(value);
                        },
                        activeColor: Colors.blue,
                        inactiveThumbColor: Colors.grey,
                      ),
                    ),

                    // Time Picker
                    if (context.watch<UserProvider>().reminder) ...[
                      Padding(
                        padding: EdgeInsets.symmetric(vertical: 20),
                        child: Text(
                          S.of(context).timePickerDescription,
                          style: TextStyle(
                            color: Colors.grey,
                            fontSize: 16,
                          ),
                        ),
                      ),
                      Center(
                        child: WheelTimePicker(
                          setHours: setHours,
                          setMinutes: setMinutes,
                          initialHours:
                              context.read<UserProvider>().reminderHours,
                          initialMinutes:
                              context.read<UserProvider>().reminderMinutes,
                        ),
                      ),
                    ],
                  ],
                ),
              ),

              // Arrow Button (only shown when enabled)
              Padding(
                padding: const EdgeInsets.only(bottom: 30),
                child: Align(
                    alignment: Alignment.bottomRight,
                    child: SizedBox(
                      width: 50,
                      height: 50,
                      child: IconButton(
                        onPressed: onPress,
                        icon: Icon(Icons.arrow_forward), // Arrow icon
                        iconSize: 30,
                        color: HexColor(dark),
                        style: ButtonStyle(
                            backgroundColor:
                                WidgetStatePropertyAll<Color>(Colors.white)),
                      ),
                    )),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
